cardano-cli/src/Cardano/CLI/EraBased/Options/Genesis.hs

```diff
@@ -16,61 +16,78 @@ import           Cardano.CLI.EraBased.Options.Common
 import           Cardano.CLI.Parser
 import           Cardano.CLI.Types.Common
 
-import           Data.Foldable
-import           Data.Maybe (fromMaybe)
+import           Data.Maybe
 import           Data.Word (Word64)
 import           Options.Applicative hiding (help, str)
 import qualified Options.Applicative as Opt
 
 {- HLINT ignore "Use <$>" -}
 {- HLINT ignore "Move brackets to avoid $" -}
 
-pGenesisCmds :: EnvCli -> Parser (GenesisCmds era)
+pGenesisCmds :: ()
+  => EnvCli
+  -> Maybe (Parser (GenesisCmds era))
 pGenesisCmds envCli =
-  asum
-    [ subParser "key-gen-genesis"
+  subInfoParser "genesis"
+    ( Opt.progDesc
+        $ mconcat
+          [ "Genesis block commands."
+          ]
+    )
+    [ Just
```

Comment:
Why the change from `asum`?

Reply:
`asum` is in the implementation of `subInfoParser`:

```
subInfoParser :: String -> InfoMod a -> [Maybe (Parser a)] -> Maybe (Parser a)
subInfoParser name i mps = case catMaybes mps of
  [] -> Nothing
  parsers -> Just $ subParser name $ Opt.info (asum parsers) i
```

`subInfoParser` is used instead so that commands are self-describing.  Previously the implementation of the parent command group provided the description.